third_party/txt/src/txt/font_collection.cc

```diff
@@ -202,6 +202,12 @@ std::shared_ptr<minikin::FontFamily> FontCollection::CreateMinikinFontFamily(
 const std::shared_ptr<minikin::FontFamily>& FontCollection::MatchFallbackFont(
     uint32_t ch,
     std::string locale) {
+  // Check if the ch's matched font has been cached. We cache the results of
+  // this method as repeated matchFamilyStyleCharacter calls can become
+  // extremely laggy when typing a large number of complex emojis.
+  if (fallback_match_cache_.count(ch) > 0) {
```

Comment:
This will do two lookups in the map.  Save a lookup by calling `find` and then checking the resulting iterator.